src/webnn_native/openvino/GraphIE.cpp

```diff
@@ -173,27 +173,49 @@ namespace webnn_native { namespace ie {
             return transposeNode;
         }
 
-        // hwio => oihw or ohwi => oihw
+        // FilterOperandLayout => oihw or FilterOperandLayout => iohw
         ngraph_node_t* TransposeFilterLayout(const ngraph_node_t* node,
-                                             ml::FilterOperandLayout layout) {
-            if (layout == ml::FilterOperandLayout::Oihw) {
-                return const_cast<ngraph_node_t*>(node);
-            }
-
+                                             ml::FilterOperandLayout layout,
+                                             bool transpose = false) {
             std::vector<int64_t> order;
-            switch (layout) {
-                case ml::FilterOperandLayout::Hwio:
-                    order = std::vector<int64_t>{3, 2, 0, 1};
-                    break;
-                case ml::FilterOperandLayout::Ohwi:
-                    order = std::vector<int64_t>{0, 3, 1, 2};
-                    break;
-                case ml::FilterOperandLayout::Ihwo:
-                    order = std::vector<int64_t>{3, 0, 1, 2};
-                    break;
-                default:
-                    WEBNN_ASSERT(0, "The filter layout isn't supported.");
-                    break;
+            if (transpose) {
+                // FilterOperandLayout => iohw
+                switch (layout) {
+                    case ml::FilterOperandLayout::Oihw:
+                        order = std::vector<int64_t>{1, 0, 2, 3};
+                        break;
+                    case ml::FilterOperandLayout::Hwio:
+                        order = std::vector<int64_t>{2, 3, 0, 1};
+                        break;
+                    case ml::FilterOperandLayout::Ohwi:
+                        order = std::vector<int64_t>{3, 0, 1, 2};
+                        break;
+                    case ml::FilterOperandLayout::Ihwo:
+                        order = std::vector<int64_t>{0, 3, 1, 2};
+                        break;
+                    default:
+                        WEBNN_ASSERT(0, "The filter layout isn't supported.");
+                        break;
+                }
+            } else {
+                // FilterOperandLayout => oihw
+                if (layout == ml::FilterOperandLayout::Oihw) {
```

Comment:
Move it to L.205
```
case ml::FilterOperandLayout::Oihw:
    return const_cast<ngraph_node_t*>(node);
```

Reply:
Optimized and rebased the branch onto main.